views/forms/TextInputTopLabel/index.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import { OutlinedInput, InputLabel } from '@mui/material/'
+import { useController } from 'react-hook-form'
+
+const TextInputTopLabel = (props) => {
+  const { errors } = props
+  const { field } = useController(props)
+
+  return (
+    <>
+      <InputLabel
+        aria-label={props.label}
+        shrink
+        {...props.inputLabel}
+        htmlFor={props.name}
+      >
+        {props.label}
+      </InputLabel>
+      <OutlinedInput
+        {...field}
+        aria-invalid={!!errors ? 'true' : 'false'}
+        id={props.name}
+        fullWidth={props.fullWidth}
+        error={!!errors}
+        inputProps={{ ...props.inputProps }}
+        margin={props.margin || 'none'}
+        required={props.required}
+        type={props.type}
+        sx={props.sx}
+        size={props.size}
+      />
+      {!!errors?.message && (
+        <InputLabel
+          aria-label={`${props.name}-error`}
+          error
+          {...props.errorStyleProps}
```

Comment:
should this component know how to style errors so the parent doesn't need to tell it?

Reply:
0730f7e